R/evaluate.R

```diff
@@ -106,6 +106,63 @@ tf_evaluate.tfb <- function(object, ..., arg) {
   ret
 }
 
+#' @export
+#' @rdname tf_evaluate
+tf_evaluate.tfb_wavelet <- function(object, ..., arg) {
+  if (missing(arg)) {
+    if (nargs() == 1) return(tf_evaluations(object))
+    if (nargs() == 2) arg <- list(...)[[1]]
+    if (nargs() > 2) {
+      stop("too many arguments - can't use ...")
+    }
+  }
+  if (is.null(arg)) return(tf_evaluations(object))
+  arg <- ensure_list(arg)
+  assert_arg(arg, object, check_unique = FALSE)
+  coef_mat <- do.call(cbind, coef(object))
+  if (length(arg) == 1) {
+    arg <- unlist(arg)
+    evals <- evaluate_tfb_once(
+      x = arg,
+      arg = tf_arg(object),
+      coefs = coef_mat[-nrow(coef_mat),],
+      basis = attr(object, "basis"),
+      X = attr(object, "basis_matrix"),
+      resolution = tf_resolution(object)
+    )
+    
+    ret <- if (length(arg) == 1) {
+      # avoid cast to simple vector for point evaluations
+      split(evals, seq_along(evals))
+    } else {
+      split(evals, col(as.matrix(evals)))
+    }  
+    
+    ret <- pmap(list(x = ret,
+                     y = attr(object, "slope_params"), 
+                     z = coef_mat[nrow(coef_mat),]),
+                function(x, y, z) {
+                  func <- y[1] + y[2] * arg
+                  as.vector(x + as.matrix(func) %*% z)
+                })
```

Comment:
i don't really like this, this seems overly complicated to me. 
if we simply append the basis matrix for wavelets by one column for the intercept and one column for the slope over t (i.e., simply a column with the `argvals`), then we could get rid of this additional "slope_params" complication entirely, save the intercept and slope as entries in the coefficient vector fo each function, and just use evaluate_tfb_once here, without this extra call to pmap etc.... so basis matrix should simply be something like `cbind(1, argvals, ZDaub(argvals, <options>))`